_ml-commons-plugin/model-serving-framework.md

```diff
@@ -0,0 +1,249 @@
+  "state" : "COMPLETED",
+  "worker_node" : "KzONM8c8T4Od-NoUANQNGg",
+  "create_time" : 3455961564003,
+  "last_update_time" : 3216361373241,
+  "is_async" : true
+}
+```
+
+### Load model from index
+
+With the `model_id`, you can now load the model from the model's index, then create an instance of the model to save in an ML node's available cache.
+
+Add the `model_id` to the load API operation. 
+
+```json
+POST /_plugins/_ml/models/<model_id>/_load
+```
+
+#### Request: Load into any available ML node
+
+In this example request, OpenSearch loads the model into any available OpenSearch node:
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_load
+```
+
+#### Request: Load into a specific node
+
+If you want to reserve the memory of other ML nodes within your cluster, you can load your model into a specific node(s) by specifying the `node_id` in the request body:
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_load
+{
+    "node_ids": ["4PLK7KJWReyX0oWKnBA8nA"]
+}
+```
+
+#### Response
+
+All models load asynchronously. Therefore, the load API responds a new `task_id` based on the load, and returns a `status` for the task.
+
+```json
+{
+  "task_id" : "hA8P44MBhyWuIwnfvTKP",
+  "status" : "CREATED"
+}
+```
+
+### Check model load status
+
+With your `task_id` from the load response, you can use the ``GET _ml/tasks` API to see the loading status of your model. Before a loaded model can be used for inferences, the load tasks `state` must show as `COMPLETED`. 
+
+#### Request
+
+```json
+GET /_plugins/_ml/tasks/hA8P44MBhyWuIwnfvTKP
+```
+
+#### Response
+
+```json
+{
+  "model_id" : "WWQI44MBbzI2oUKAvNUt",
+  "task_type" : "LOAD_MODEL",
+  "function_name" : "TEXT_EMBEDDING",
+  "state" : "COMPLETED",
+  "worker_node" : "KzONM8c8T4Od-NoUANQNGg",
+  "create_time" : 1665961803150,
+  "last_update_time" : 1665961815959,
+  "is_async" : true
+}
+```
+
+## Use loaded model for inferences
+
+Use the predict operation to perform inferences with your loaded model. To streamline the prediction process, loaded models can call their functions from memory with the `model_id`. 
+
+```json
+POST /_plugins/_ml/models/<model_id>/_predict
+```
+
+With model loaded, OpenSearch can then output inference results based on the loaded model.
+
+### Request
+
+```json
+POST /_plugins/_ml/models/WWQI44MBbzI2oUKAvNUt/_predict
+{
+  "text_docs":[ "today is sunny"],
+  "return_number": true,
+  "target_response": ["sentence_embedding"]
+}
+```
+
+### Response
+
+```json
+{
+  "inference_results" : [
+    {
+      "output" : [
+        {
+          "name" : "sentence_embedding",
+          "data_type" : "FLOAT32",
+          "shape" : [
+            384
+          ],
+          "data" : [
+            -0.023315024,
+            0.08975691,
+            0.078479774,
+            ...
+          ]
+        }
+      ]
+    }
+  ]
+}
+```
+
+## Unload the model
+
+To unload a model from memory, use the unload operation.
```

Comment:
If user don't need to run predict for some model, they can use unload API to close model instance and remove it from cache. Unload API won't delete model from model index.

Reply:
How does this sound? 

```suggestion
If you're done making predictions with your model, use the unload operation to remove the model from your memory cache. The model will remain accessible in the model index.
```

Reply:
How about we say we "remove the model from the ML Node(s) to free up resources." I would remove the word "cache" because it's not quite accurate.